ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MicrosoftSentinelForm.tsx

```diff
@@ -0,0 +1,529 @@
+
+import IntegrationFormActions from '../IntegrationFormActions';
+import FormLabelGroup from '../FormLabelGroup';
+
+export type MicrosoftSentinel = {
+    microsoftSentinel: {
+        logIngestionEndpoint: string;
+        directoryTenantId: string;
+        applicationClientId: string;
+        clientCertAuthConfig: {
+            clientCert: string;
+            privateKey: string;
+        };
+        secret: string;
+        alertDcrConfig: {
+            dataCollectionRuleId: string;
+            streamName: string;
+            enabled: boolean;
+        };
+        auditLogDcrConfig: {
+            dataCollectionRuleId: string;
+            streamName: string;
+            enabled: boolean;
+        };
+    };
+    type: 'microsoftSentinel';
+} & NotifierIntegrationBase;
+
+export type MicrosoftSentinelFormValues = {
+    notifier: MicrosoftSentinel;
+    updatePassword: boolean;
+};
+
+export const validationSchema = yup.object().shape({
+    notifier: yup.object().shape({
+        name: yup.string().trim().required('Email integration name is required'),
+        microsoftSentinel: yup.object().shape({
+            logIngestionEndpoint: yup
+                .string()
+                .trim()
+                .required('A log ingestion endpoint is required'),
+            directoryTenantId: yup.string().trim().required('A directory tenant ID is required'),
+            applicationClientId: yup
+                .string()
+                .trim()
+                .required('A application client ID is required'),
+            secret: yup.string().trim(),
+            clientCertAuthConfig: yup.object().shape({
+                clientCert: yup.string().trim(),
+                privateKey: yup.string().trim(),
+            }),
+        }),
+    }),
+    updatePassword: yup.bool(),
+});
+
+export const defaultValues: MicrosoftSentinelFormValues = {
+    notifier: {
+        id: '',
+        name: '',
+        microsoftSentinel: {
+            logIngestionEndpoint: '',
+            directoryTenantId: '',
+            applicationClientId: '',
+            clientCertAuthConfig: {
+                clientCert: '',
+                privateKey: '',
+            },
+            secret: '',
+            alertDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+            auditLogDcrConfig: {
+                dataCollectionRuleId: '',
+                streamName: '',
+                enabled: false,
+            },
+        },
+        labelDefault: '',
+        labelKey: '',
+        uiEndpoint: window.location.origin,
+        type: 'microsoftSentinel',
+    },
+    updatePassword: true,
+};
+
+function MicrosoftSentinelForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<MicrosoftSentinel>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
```

Comment:
Nit: the object spread will create a shallow copy of the module-level `defaultValues` here, which could result in data overwriting the defaults and being applied again on follow up visits to the form. I'd recommend we avoid this with `structuredClone` or lodash's `cloneDeep` to get a totally new object as the initial value.

Reply:
Van, here is the pattern to follow:
https://github.com/stackrox/stackrox/pull/12874/files#diff-e6973e0cce8fcb46bd43624a8578812f1dca98cae2589b99edfe2f4656653872L123-L136